nixops/backends/__init__.py

```diff
@@ -247,16 +247,29 @@ def reboot_sync(self, hard: bool = False) -> None:
         """Reboot this machine and wait until it's up again."""
         self.reboot(hard=hard)
         self.log_start("waiting for the machine to finish rebooting...")
-        nixops.util.wait_for_tcp_port(
-            self.get_ssh_name(),
-            self.ssh_port,
-            open=False,
-            callback=lambda: self.log_continue("."),
+
+        nixops.util.wait_for_fail(
+            fn=lambda: self.ssh.run_command(
+                ["true"],
+                user=self.ssh_user,
+                forward_agent=False,
+                logged=False,
+                connection_tries=1,
+            )
```

Comment:
Let's wrap this in a convenience method of like `is the machine actually reachable?`